### ISSUE TITLE
Merge b2g branch into dev

### ISSUE THREAD

**Issue by jedp** [mozilla/persona#2982]
The work done to support marketplace and native implementations (b2g) should be merged into dev.

Once this is done, we can rejoice, as all users will share a single database, and RPs will not need to know in advance whether they are serving native or shimmed; they will only need to include js from persona.org.

This needs to happen before the final b2g code is shipped to vendors to put on phones that go to market.  I think we should plan on anticipating this date by about a month to allow for stress-free QA, staging, and deployment.

Bug [836928](https://bugzilla.mozilla.org/show_bug.cgi?id=836928) tracks this on the bugzilla side.

**Comment by lloyd** [comment]
Is there any creative way we can make this transparent to rps?

Isn't this analogous to the browserid-> persona transition?

-- lloyd (thumb-typing)

**Comment by jaredhirsch** [comment]
@jedp do we have a hard date when this needs to be all merged, happy, and
the train deployed?

**Comment by jedp** [comment]
Well, the closest thing we have to a hard date is the time when FirefoxOS phones hit the market, which will be this summer.

**Comment by jedp** [comment]
Preliminary PR https://github.com/mozilla/browserid/pull/3059 - still work in progress

**Comment by jedp** [comment]
So ... close ...

**Comment by jaredhirsch** [comment]
@jedp good to close?

**Comment by shane-tomlinson** [comment]
Closing
